plugins/qcheck-stm/src/ortac_qcheck_stm.ml

```diff
@@ -4,9 +4,12 @@ module Ir_of_gospel = Ir_of_gospel
 module Reserr = Reserr
 module Stm_of_ir = Stm_of_ir
 
-let main path init sut () =
+let main path init sut output () =
   let open Reserr in
-  let pp = Fmt.((pp Ppxlib_ast.Pprintast.structure) stdout) in
+  let channel =
+    Registration.get_channel output |> Format.formatter_of_out_channel
```

Comment:
While reviewing this, I notice that all plugins are actually using a formatter. Maybe `Registration` could provide a `get_out_formatter` rather than a `get_channel`?

Reply:
By the way, I find a formatter called `channel` a bit misleading...

Reply:
`get_out_formatter` is a good idea.
Added in a third commit to the PR.